### ISSUE TITLE
feature request about resuming model training with epochs change

### ISSUE THREAD

**Issue by yantaozhao** [ultralytics/ultralytics#19467]
### Scene

given trained model total 100 epochs done, then I want to resume training based on previous trained model and increasing/decreasing some epochs.

first 100 epochs:
```python
model = YOLO("yolo.pt")
results = model.train(
    data="datasets/xxx.yaml",
    save_period=2,
    epochs=100,
)
```

then, resume training and change epochs:
```python
model = YOLO("epoch90.pt")
results = model.train(
    data="datasets/xxx.yaml",
    save_period=2,
    epochs=300,  # more epochs
    resume=True, # resume flag
)
```

but `epochs=300` not works, it still stops at 100.

As already pointed in https://github.com/ultralytics/ultralytics/issues/16402 and https://github.com/ultralytics/ultralytics/issues/18154 . I'm wondering why changing epochs when resuming is not supported in current code base?

Is it possible to support it?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @yantaozhao, thank you for your interest in Ultralytics 🚀! We recommend exploring the [Docs](https://docs.ultralytics.com/) for comprehensive guides, where you can find examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) use cases. These may already address many common questions.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that includes your full code and details about your environment. This will enable us to reproduce the issue and assist you more effectively.

For your ❓ Question, it seems you're looking to extend training beyond 100 epochs. To help clarify or debug the behavior you're experiencing, could you confirm:
1. The exact version of the `ultralytics` package you are using? You can check with `pip show ultralytics`.
2. Whether you've upgraded to the latest version using:
   ```bash
   pip install -U ultralytics
   ```
3. The contents of your `results` or any logs generated during training.

If you believe an issue persists, please provide more detailed logs or any errors you're encountering to help us assist further.

Join the Ultralytics community where it suits you best! For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) and share your experiences 🐾.

## Upgrade

Ensure you're running the latest version of the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/), and update if necessary:
```bash
pip install -U ultralytics
```

## Environments

You can run YOLO in several pre-configured environments (all dependencies included, such as [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help you get started. One of our engineers will review your issue and provide additional guidance as soon as possible. Thank you for your patience 🙏!

**Comment by Y-T-G** [comment]
The LR scheduling is based on the number of epochs set initially and you can't change the epochs without affecting it which in turn affects the training process. If you're expecting the same loss trend to continue after increasing the epochs, then that's not the case.

https://github.com/ultralytics/ultralytics/issues/802#issuecomment-2404251798
